dask_cuda/benchmarks/local_cudf_merge.py

```diff
@@ -282,104 +214,56 @@ def main(args):
-    )
-    print_separator(separator="=")
+    print_key_value(key="Data processed", value=f"{format_bytes(results[0][0])}")
     if args.markdown:
         print("\n```")
 
-    if args.plot is not None:
-        plot_benchmark(t_runs, args.plot, historical=True)
-
-    if args.backend == "dask":
-        if args.markdown:
-            print("<details>\n<summary>Worker-Worker Transfer Rates</summary>\n\n```")
-        print_key_value(key="(w1,w2)", value="25% 50% 75% (total nbytes)")
-        print_separator(separator="-")
-        for (d1, d2), bw in sorted(bandwidths.items()):
-            key = (
-                f"({d1},{d2})"
-                if args.multi_node or args.sched_addr
-                else f"({d1:02d},{d2:02d})"
-            )
-            print_key_value(
-                key=key, value=f"{bw[0]} {bw[1]} {bw[2]} ({total_nbytes[(d1, d2)]})"
-            )
-        if args.markdown:
-            print("```\n</details>\n")
-
-    if args.benchmark_json:
-        bandwidths_json = {
-            "bandwidth_({d1},{d2})_{i}"
-            if args.multi_node or args.sched_addr
-            else "(%02d,%02d)_%s" % (d1, d2, i): parse_bytes(v.rstrip("/s"))
-            for (d1, d2), bw in sorted(bandwidths.items())
-            for i, v in zip(
-                ["25%", "50%", "75%", "total_nbytes"],
-                [bw[0], bw[1], bw[2], total_nbytes[(d1, d2)]],
-            )
-        }
-
-        with open(args.benchmark_json, "a") as fp:
-            for data_processed, took in took_list:
-                fp.write(
-                    dumps(
-                        dict(
-                            {
-                                "backend": args.backend,
-                                "merge_type": args.type,
-                                "rows_per_chunk": args.chunk_size,
-                                "base_chunks": args.base_chunks,
-                                "other_chunks": args.other_chunks,
-                                "broadcast": broadcast,
-                                "protocol": args.protocol,
-                                "devs": args.devs,
-                                "device_memory_limit": args.device_memory_limit,
-                                "rmm_pool": not args.disable_rmm_pool,
-                                "tcp": args.enable_tcp_over_ucx,
-                                "ib": args.enable_infiniband,
-                                "nvlink": args.enable_nvlink,
-                                "data_processed": data_processed,
-                                "wall_clock": took,
-                                "throughput": data_processed / took,
-                            },
-                            **bandwidths_json,
-                        )
-                    )
-                    + "\n"
-                )
+    data_processed, durations = zip(*results)
+    print_throughput_bandwidth(
+        args, durations, data_processed, p2p_bw, address_to_index
+    )
```

Comment:
The per-benchmark specific part of the pretty-printing is now just printing the configuration and then the bandwidth printing is factored into a utility function.